models/learner.py

```diff
@@ -44,35 +50,37 @@ class BaseLearner(object):
     :param state_dim: (int)
     :param batch_size: (int)
     :param seed: (int)
-    :param cuda: (bool)
+    :param cuda: (int) (default -1, CPU) equi to CUDA_VISIBLE_DEVICES
     """
 
-    def __init__(self, state_dim, batch_size, seed=1, cuda=False):
+    def __init__(self, state_dim, batch_size, seed=1, cuda=-1):
         super(BaseLearner, self).__init__()
         self.state_dim = state_dim
         self.batch_size = batch_size
-        self.model = None
+        self.module = None
         self.seed = seed
         self.use_dae = False
+        assert not self.use_dae, "Not implemented error."
         # Seed the random generator
         np.random.seed(seed)
-        th.manual_seed(seed)
-        if cuda:
+        torch.manual_seed(seed)
+        if cuda >= 0:  # This will increase the training time by 5-10%.
```

Comment:
Maybe it helps speed up in the case of a multi-gpu setting ? 
Not clear, misleading...

Reply:
I wanted to say, by making cuda deterministic would **decrease** the performance, and **increase** the training time by 5-10%. These lines (lines 67-70) can be comment-out if the user want further speed-up.